.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,160 @@
+  // TODO(josephperrott): Load this from the actual repository.
+  setConfig(<{pullRequest: PullRequestConfig; github: GithubConfig}>{
+    github: {
+      mainBranchName: 'main',
+      owner: repo.owner,
+      name: repo.repo,
+    },
+    pullRequest: {
+      githubApiMerge: false,
+    },
+  });
+  /** The configuration used for the ng-dev tooling. */
+  const config = await getConfig([assertValidGithubConfig, assertValidPullRequestConfig]);
+
+  AuthenticatedGitClient.configure(token);
+  /** The git client used to perform actions. */
+  const git = await AuthenticatedGitClient.get();
+
+  /** The pull request after being retrieved and validated. */
+  const pullRequest = await loadAndValidatePullRequest(
+    {git, config},
+    pr,
+    new PullRequestValidationConfig(),
+  );
+  core.info('Validated PR information:');
+  core.info(JSON.stringify(pullRequest));
+  /** Whether any fatal validation failures were discovered. */
+  let hasFatalFailures = false;
+  /** The status information to be pushed as a status to the pull request. */
+  let statusInfo: CommmitStatus = await (async () => {
+    // Log validation failures and check for any fatal failures.
+    if (pullRequest.validationFailures.length !== 0) {
+      core.info(`Found ${pullRequest.validationFailures.length} failing validation(s)`);
+      for (const failure of pullRequest.validationFailures) {
+        hasFatalFailures = !failure.canBeForceIgnored || hasFatalFailures;
+        await core.group('Validation failures', async () => {
+          core.info(failure.message);
+        });
+      }
+    }
+
+    // With any fatal failure the check is not necessary to do.
+    if (hasFatalFailures) {
+      core.info('One of the validations was fatal, setting the status as pending for the pr');
+      return {
+        description: 'Waiting to check mergeability due to failing status(es)',
+        state: 'pending',
+      };
+    }
+
+    try {
+      git.run(['checkout', 'main']);
+      /**
+       * A merge strategy used to perform the merge check.
+       * Any concrete class implementing MergeStrategy is sufficient as all of our usage is
+       * defined in the abstract base class.
+       * */
+      const strategy = new AutosquashMergeStrategy(git);
+      await strategy.prepare(pullRequest);
+      await strategy.check(pullRequest);
+      core.info('Merge check passes, setting a passing status on the pr');
+      return {
+        description: `Merges cleanly to ${pullRequest.targetBranches.join(', ')}`,
+        state: 'success',
+      };
+    } catch (e) {
+      // As the merge strategy class will express the failures during checks, any thrown error is a
+      // failure for our merge check.
+      let description: string;
+      if (e instanceof MergeConflictsFatalError) {
+        core.info('Merge conflict found');
+        description = `Unable to merge into ${e.failedBranches.join(
+          ', ',
+        )} please update changes or PR target`;
```

Comment:
Note: GitHub doesn't show large descriptions and just cuts them off- without even showing the full description on hover (at least on my end)

Reply:
Hmm okay, I will shorten this and put in a cutoff to make sure we don't cause an API failure, looks like the limit is 140 characters